File: app/src/main/java/ru/mamykin/foreignbooksreader/common/TextHashMap.java
package ru.mamykin.foreignbooksreader.common;

import android.support.annotation.Nullable;

import java.util.HashMap;
import java.util.Set;

public class TextHashMap extends HashMap<String, String> {
    @Nullable
    public String getKey(String keyPart) {
        final Set<String> mapSet = keySet();
        for (String str : mapSet) {
            if (str.contains(keyPart))
                return str;
        }
        return null;
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/common/TextHashMap.kt
package ru.mamykin.foreignbooksreader.common

import java.util.HashMap

class TextHashMap : HashMap<String, String>() {

    fun getKey(keyPart: String): String? {
        val mapSet = keys
        for (str in mapSet) {
            if (str.contains(keyPart))
                return str
        }
        return null
    }
}